Mathlib/Analysis/Complex/AbelLimit.lean

```diff
@@ -0,0 +1,210 @@
+  intro zn
+  calc
+    _ ≤ 1 * (1 - ‖z‖) := mul_le_mul_of_nonneg_right hM zn.le
+    _ = ‖(1 : ℂ)‖ - ‖z‖ := by rw [one_mul, norm_one]
+    _ ≤ _ := norm_sub_norm_le _ _
+
+theorem nhdsWithin_lt_le_nhdsWithin_stolzSet (hM : 1 < M) :
+    (𝓝[<] 1).map ofReal' ≤ 𝓝[stolzSet M] 1 := by
+  rw [← tendsto_id']
+  refine' tendsto_map' <| tendsto_nhdsWithin_of_tendsto_nhds_of_eventually_within ofReal'
+    (tendsto_nhdsWithin_of_tendsto_nhds <| ofRealCLM.continuous.tendsto' 1 1 rfl) _
+  simp only [eventually_iff, norm_eq_abs, abs_ofReal, abs_lt, mem_nhdsWithin]
+  refine' ⟨Set.Ioo 0 2, isOpen_Ioo, by norm_num, fun x hx ↦ _⟩
+  simp only [Set.mem_inter_iff, Set.mem_Ioo, Set.mem_Iio] at hx
+  simp only [Set.mem_setOf_eq, stolzSet, ← ofReal_one, ← ofReal_sub, norm_eq_abs, abs_ofReal,
+    abs_of_pos hx.1.1, abs_of_pos <| sub_pos.mpr hx.2]
+  exact ⟨hx.2, lt_mul_left (sub_pos.mpr hx.2) hM⟩
+
+end StolzSet
+
+variable {f : ℕ → ℂ} {l : ℂ} (h : Tendsto (fun n ↦ ∑ i in range n, f i) atTop (𝓝 l))
+
+/-- Auxiliary lemma for Abel's limit theorem. The difference between the sum `l` at 1 and the
+power series's value at a point `z` away from 1 can be rewritten as `1 - z` times a power series
+whose coefficients are tail sums of `l`. -/
+lemma abel_aux {z : ℂ} (hz : ‖z‖ < 1) :
+    Tendsto (fun n ↦ (1 - z) * ∑ i in range n, (l - ∑ j in range (i + 1), f j) * z ^ i)
+      atTop (𝓝 (l - ∑' n, f n * z ^ n)) := by
+  let s := fun n ↦ ∑ i in range n, f i
+  have k := h.sub (summable_power_of_norm_lt_one h.cauchySeq hz).hasSum.tendsto_sum_nat
+  simp_rw [← sum_sub_distrib, ← mul_one_sub, ← geom_sum_mul_neg, ← mul_assoc, ← sum_mul,
+    mul_comm, mul_sum _ _ (f _), range_eq_Ico, ← sum_Ico_Ico_comm', ← range_eq_Ico,
+    ← sum_mul] at k
+  conv at k =>
+    enter [1, n]
+    rw [sum_congr (g := fun j ↦ (∑ k in range n, f k - ∑ k in range (j + 1), f k) * z ^ j)
+      rfl (fun j hj ↦ by congr 1; exact sum_Ico_eq_sub _ (mem_range.mp hj))]
+  suffices : Tendsto (fun n ↦ (l - s n) * ∑ i in range n, z ^ i) atTop (𝓝 0)
+  · simp_rw [mul_sum] at this
+    replace this := (this.const_mul (1 - z)).add k
+    conv at this =>
+      enter [1, n]
+      rw [← mul_add, ← sum_add_distrib]
+      enter [2, 2, i]
+      rw [← add_mul, sub_add_sub_cancel]
+    rwa [mul_zero, zero_add] at this
+  rw [← zero_mul (-1 / (z - 1))]
+  apply Tendsto.mul
+  · simpa only [neg_zero, neg_sub] using (tendsto_sub_nhds_zero_iff.mpr h).neg
+  · conv =>
+      enter [1, n]
+      rw [geom_sum_eq (by contrapose! hz; simp [hz]), sub_div, sub_eq_add_neg, ← neg_div]
+    rw [← zero_add (-1 / (z - 1)), ← zero_div (z - 1)]
+    apply Tendsto.add (Tendsto.div_const (tendsto_pow_atTop_nhds_0_of_norm_lt_1 hz) (z - 1))
+    simp only [zero_div, zero_add, tendsto_const_nhds_iff]
+
+/-- **Abel's limit theorem**. Given a power series converging at 1, the corresponding function
+is continuous at 1 when approaching 1 within a fixed Stolz set. -/
+theorem tendsto_tsum_power_nhdsWithin_stolzSet {M : ℝ} :
```

Comment:
The suggestion to rename `power` to `powerSeries` also applies to this and the following two statements.

Reply:
We merge #9955 first and then I'll get to this.

Reply:
@MichaelStollBayreuth done.